### ISSUE TITLE
IBFEMethod handles restart data in a weird way

### ISSUE THREAD

**Issue by drwells** [IBAMR/IBAMR#698]
I spent way too long trying to figure this out and I suspect others have gotten confused by this too. `IBFEMethod`'s constructor needs the restart read directory and restart restore number as arguments: I haven't looked too closely, but (at least in the heart model) no other major object needs these as arguments.

Additionally, we need to explicitly call `IBFEMethod::writeFEDataToRestartFile` to save libMesh data. We should try to teach SAMRAI how to deal with a callback for saving non-SAMRAI restart data so that we don't need this explicit call everywhere, i.e., we should not have to write
```cpp
if (dump_restart_data && (iteration_num % restart_dump_interval == 0 || last_step))
{
    pout << "\nWriting restart files...\n\n";
    RestartManager::getManager()->writeRestartFile(restart_dump_dirname, iteration_num);
    ib_method_ops->writeFEDataToRestartFile(restart_dump_dirname, iteration_num);
}
```
I don't know if that's possible but we should try and (minimally) update all IBFE examples to actually call both functions.

**Comment by boyceg** [comment]
The difficulty that this attempts to work around is that the libMesh DOF vectors go in their own file(s), but the `Serializable` API just talks about putting things into _databases_, not _files_.

We could make a class that implements `SAMRAI::tbox::Serializable` that stores libMesh data, but I think at best it will be an ongoing maintenance chore and could make it harder to support multiple versions of libMesh.

We also could assume that the only time that we are serializing a class like `IBFEMethod` is if we are writing a restart file, and try to see if we can just store the name of the libMesh data file as a part of the `IBFEMethod` class database. There does not appear to be a mechanism to ask SAMRAI where it is writing restart files, however, so there still would be this extra piece of information that would need to be passed on --- namely, where to write the files.

**Comment by drwells** [comment]
I suspected that would be the answer. I'll spend a little bit of time poking around in SAMRAI to see what we can get, but I will be satisfied to just update the IBFE examples to make both restart function calls and improve some documentation.

**Comment by drwells** [comment]
@marshallrdavey Would you be willing to work on this for the release? There are two parts:
1. Update the documentation of IBFEMethod to describe how to save restart data
2. Update the IBFE examples to actually save restart data

**Comment by drwells** [comment]
I talked about this with @marshallrdavey earlier today. Since this is just a documentation fix I'm okay with adding it late in the release process (i.e., near the 20th). @marshallrdavey can you commit to finishing this by the 20th?

**Comment by marshallrdavey** [comment]
@drwells I will do my best!

**Comment by drwells** [comment]
I'll hold you to it!

**Comment by drwells** [comment]
We still need to update the examples.

**Comment by drwells** [comment]
Fixed by #805.
